scheduler_test.go

```diff
@@ -631,6 +631,19 @@ func TestScheduler_Stop(t *testing.T) {
 		s.Stop()
 		assert.False(t, s.IsRunning())
 	})
+	t.Run("waits for jobs to finish processing before returning .Stop()", func(t *testing.T) {
+		t.Parallel()
+		i := 0
+		s := NewScheduler(time.UTC)
+		s.Every(1).Second().Do(func() {
+			time.Sleep(2 * time.Second)
+			i = 1
```

Comment:
this is creating a race condition. you could as atomic int, or mutex for I here, or our semaphore pattern

Reply:
or no, this has to do with the stop method